specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-09-01-preview/inference.json

```
@@ -0,0 +1,1443 @@
+            "schema": {
+              "$ref": "#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "MachineLearningInferenceEndpoints"
+        ],
+        "description": "Creates or updates an Inference Endpoint. This call will update an Inference Endpoint if it exists. This is a nonrecoverable operation. If your intent is to create a new Inference Endpoint, do a GET first to verify that it does not exist yet.",
+        "operationId": "MachineLearningInferenceEndpoint_CreateOrUpdate",
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Create Or Update Endpoint": {
+            "$ref": "./examples/createOrUpdateInferenceEndpoint.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/APIVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/InferenceEndpointNameParameter"
+          },
+          {
+            "in": "body",
+            "name": "properties",
+            "description": "The payload that is used to create or update the Inference Endpoint.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/InferenceEndpointCreateRequest"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Inference Endpoint creation or update initiated.",
+            "schema": {
+              "$ref": "#/definitions/InferenceEndpointResource"
+            }
+          },
+          "202": {
+            "description": "Inference Endpoint creation or update initiated.",
+            "headers": {
+              "Azure-AsyncOperation": {
+                "description": "URI to poll for asynchronous operation status.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/MachineLearningServiceError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/inferenceEndpoints/locations/{locationName}/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "MachineLearningInferenceEndpoints"
+        ],
+        "x-ms-examples": {
+          "Check Endpoint Name": {
+            "$ref": "./examples/checkInferenceEndpointName.json"
+          }
+        },
+        "operationId": "MachineLearningInferenceEndpoint_CheckNameAvailability",
```

Align with the List operationId
```suggestion
        "operationId": "MachineLearningInferenceEndpoints_CheckNameAvailability",
```